staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -336,6 +336,15 @@ func InClusterConfig() (*Config, error) {
 	}, nil
 }
 
+type notInCluster struct{ error }
+
+// IsNotInCluster checks whether the error means that this process is not in a in-cluster context with
+// KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT set as environment variables.
+func IsNotInCluster(err error) bool {
+	_, ok := err.(notInCluster)
```

Comment:
what about do a type switch? and we can handle nil check here

Reply:
see @awly suggestion https://github.com/kubernetes/kubernetes/pull/67545#discussion_r213117954. That's even simpler.